samples/nrf9160/mqtt_simple/src/main.c

```diff
@@ -509,6 +510,28 @@ static void button_handler(uint32_t button_states, uint32_t has_changed)
 }
 #endif
 
+#if defined(CONFIG_SHELL)
+static int shell_publish(const struct shell *shell, size_t argc, char **argv)
+{
+	ARG_UNUSED(argc);
+	ARG_UNUSED(argv);
+
+	int ret;
+
+	ret = data_publish(&client,
+				MQTT_QOS_1_AT_LEAST_ONCE,
+				CONFIG_BUTTON_EVENT_PUBLISH_MSG,
+				sizeof(CONFIG_BUTTON_EVENT_PUBLISH_MSG)-1);
+	if (ret) {
+		LOG_ERR("Publish failed: %d", ret);
+	}
+
+	return ret;
+}
+
+SHELL_CMD_REGISTER(publish, NULL, "Publish data to configured publish topic", shell_publish);
```

Comment:
Would it make sense to register a `mqtt` command, an make `publish` its subcommand? IMO `mqtt publish` would be more meaningful.